searchcore/src/vespa/searchcore/proton/server/disk_mem_usage_sampler.cpp

```diff
@@ -46,12 +49,24 @@ DiskMemUsageSampler::setConfig(const Config &config)
                                         maxInterval, maxInterval);
 }
 
+bool
+DiskMemUsageSampler::isSampleValid(uint64_t diskUsage, const TransientResourceUsage & transientUsage) const {
+    bool isLargeDiskUsageDrop = diskUsage + 1_Gi < _lastDiskUsage;
+    bool isLargeTransientDiskUsageDrop = transientUsage.disk() + 1_Gi < _lastTransientUsage.disk();
+    return isLargeDiskUsageDrop == isLargeTransientDiskUsageDrop;
+}
 void
 DiskMemUsageSampler::sampleUsage()
 {
     sampleMemoryUsage();
-    sampleDiskUsage();
-    sample_transient_resource_usage();
+    uint64_t diskUsage = sampleDiskUsage();
+    TransientResourceUsage transientDiskUsage = sample_transient_resource_usage();
+    if (isSampleValid(diskUsage, transientDiskUsage)) {
+        _filter.setDiskUsedSize(diskUsage);
+        _filter.set_transient_resource_usage(transientDiskUsage);
+        _lastDiskUsage = diskUsage;
```

Comment:
Consider always updating `_lastDiskUsage` and `_lastTransientUsage`, to avoid case where future samples are always considered invalid (e.g. if field with attribute vector aspect was removed, dropping disk usage by more than 1 GB ).

Reply:
PR redone to avoid remembering state.